projects/02-meme-generator-api/solution/lib/memeapi.rb

```diff
@@ -0,0 +1,37 @@
+require 'sinatra/base'
+require './lib/download'
+require './lib/request_image'
+require 'pry'
+require './lib/validator'
+
+class MemeApi < Sinatra::Application
+
+  class << self
+    def run
+      run!
+    end
+  end
+
+  post '/memes' do
+    #req = RequestImage.new(request.body.read)
+    #halt req.status, req.message if req.status == 400
+    #image_name = req.create_image
+    json_body = JSON.parse(request.body.read)
+    validation = Validator.validate_image(json_body)
+    unless validation
+      halt 400, validation.message
```

Comment:
Once rspec parsing is fixed, i got confirmation that this line is never executed, and the code that expects a 400 error is actually getting a 500 error.

validate_image raises an error in case of any validation failures, it does not return a result with message on it.

When raising errors, you need to rescue to catch them, as you did for Download::Error